apps/meteor/server/startup/migrations/v278.ts

```diff
@@ -0,0 +1,45 @@
+import { BannerPlatform } from '@rocket.chat/core-typings';
+import { Users, Settings } from '@rocket.chat/models';
+
+import { addMigration } from '../../lib/migrations';
+import { isEnterprise } from '../../../ee/app/license/server';
+import { Banner } from '../../sdk';
+import { settings } from '../../../app/settings/server';
+
+addMigration({
+	version: 278,
+	up() {
+		const query = {
+			_id: { $in: [/^Accounts_OAuth_Custom-?([^-_]+)$/] },
+			value: true,
+		};
+
+		const isCustomOAuthEnabled = !!Promise.await(Settings.findOne(query));
+		const LDAPEnabled = settings.get('LDAP_Enable');
+		const SAMLEnabled = settings.get('SAML_Custom_Default');
+
+		const isEE = isEnterprise();
+
+		if (!isEE && (isCustomOAuthEnabled || LDAPEnabled || SAMLEnabled)) {
+			return;
+		}
+
+		const users = Promise.await(Users.find().toArray());
+
+		const banners = users.map((a) => Promise.await(Banner.getBannersForUser(a._id, BannerPlatform.Web))).flat();
```

Comment:
we also should avoid queries in loops like whenever it is possible.. in this case, after fetching 500k+ users, a new query will be performed for every user, synchronously (due to `Promise.await`).. so let's say the query takes only 20ms to execute, for 500k users this loop would take at least 10.000 seconds.